### PR TITLE
fix: reenable macos builds

### PR DESCRIPTION
**Pull request by romange**
Also, add debug function that prints local state if deadlocks occure.

### CODE REVIEW

**Review by dranikpg — APPROVED** [review]
We already have all this info https://github.com/dragonflydb/dragonfly/blob/20bda84317a21336bcd7ae2ea52f2ee1bab0ab94/src/server/test_utils.cc#L239-L249

**Comment by romange** [comment]
> We already have all this info
> 
> https://github.com/dragonflydb/dragonfly/blob/20bda84317a21336bcd7ae2ea52f2ee1bab0ab94/src/server/test_utils.cc#L239-L249

It's not the same though. Here I check if the callback have run because then it would point to a bug in helio (condition was reached but the barrier has not been woken up).

**Review by dranikpg — DISMISSED** [review]

**Review by dranikpg — DISMISSED** [review]

File: src/server/tiered_storage_test.cc [hunk 1]
```diff
@@ -310,7 +310,10 @@ TEST_F(TieredStorageTest, MemoryPressure) {
   constexpr size_t kNum = 10000;
   for (size_t i = 0; i < kNum; i++) {
     auto resp = Run({"SET", absl::StrCat("k", i), BuildString(10000)});
-    ASSERT_EQ(resp, "OK") << i;
+    if (resp != "OK"sv) {
+      resp = Run({"INFO", "ALL"});
+      ASSERT_FALSE(true) << i << "\nInfo ALL:\n" << resp.GetString();
+    }
     // TODO: to remove it once used_mem_current is updated frequently.
```
Comment (dranikpg) [inline src/server/tiered_storage_test.cc:317]: Might have been me who introduced flakiness here 🫣

**Review by dranikpg — DISMISSED** [review]

File: src/server/tiered_storage_test.cc (hunk 1 above)
Comment (romange) [reply src/server/tiered_storage_test.cc:317]: how?

Comment (dranikpg) [reply src/server/tiered_storage_test.cc:317]: I remember it failed for me once when I ran it for my cool entry refactoring PR

**Review by dranikpg — APPROVED** [review]

**Pull request merged**